agent/job_runner.go

```diff
@@ -570,17 +570,19 @@ func (w LogWriter) Write(bytes []byte) (int, error) {
 	return len(bytes), nil
 }
 
-func validateRepository(allowedRepos []string, pipelineRepo string) error {
-	if len(allowedRepos) == 0 {
+// validateJobValue returns an error if a job value doesn't match
+// any allowed patterns.
+func validateJobValue(allowedPatterns []string, jobValue string) error {
+	if len(allowedPatterns) == 0 {
 		return nil
 	}
 
-	for _, allowedRepo := range allowedRepos {
-		if match, _ := regexp.MatchString(allowedRepo, pipelineRepo); match {
+	for _, at := range allowedPatterns {
+		if match, _ := regexp.MatchString(at, jobValue); match {
 			return nil
 		}
 	}
-	return fmt.Errorf("repository %s isn't allowed", pipelineRepo)
+	return fmt.Errorf("%s has no match in %s", jobValue, allowedPatterns)
```

Comment:
i wonder if it'd be nice to say where the list of patterns came from - ie have the error message be something like:
```
github.com/crimes/evil-buildkite-plugin has no match in allowed plugins: [github.com/buildkite/.*]
```